scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -69,68 +116,167 @@ private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl s
         };
     }
 
-    List<Issue> runExternalAnalyzers(Project project) {
+    Optional<Map<String, List<Rule>>> getExternalAnalyzers(PrintStream outputStream) {
```

Comment:
Ideally you should not return an optional collection. Instead you should just return just an empty collection. Otherwise receiver has to handle case where value is empty and collection is empty as two distinct scenarios which could be confusing. If you really want to signal 3 states collection being empty, collection having values and "empty" then it is better to return a record with a boolean flag saying what "empty" means in addition to the collection